gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -286,5 +286,45 @@ namespace gtsam {
         AlgebraicDecisionTree<Key>(keys, table),
         cardinalities_(keys.cardinalities()) {}
 
+  /* ************************************************************************ */
+  DecisionTreeFactor::shared_ptr DecisionTreeFactor::prune(
```

Comment:
This should be a PR in GTSAM with a test

Reply:
Done https://github.com/borglab/gtsam/pull/1151